packages/core/src/odata/common/selectable/link.ts

```diff
@@ -33,10 +33,10 @@ export class Link<
    * @param link - Link to be cloned
    * @returns Newly created link
    */
-  static clone<EntityT extends EntityBase, LinkedEntityT extends EntityBase>(
-    link: Link<EntityT, LinkedEntityT>
-  ): Link<EntityT, LinkedEntityT> {
-    const clonedLink = new Link<EntityT, LinkedEntityT>(
+  static clone<EntityT1 extends EntityBase, LinkedEntityT2 extends EntityBase>(
```

Comment:
Here we use `LinkedEntityT2 `, and in this [line](https://github.com/SAP/cloud-sdk/pull/547/commits/44d6f3fbcc6ae9fe0c253dfc955dda440958eb80#diff-05764056f696813806910bf1134c5979R25) we use `LinkedEntityT1 `. Since `OneToOneLink extends Link`, I guess this is also a fix to avoid using the same method signature?

Reply:
Yes, they just need to have different names than the ones from the class.